src/main_vm/opcodes/log.rs

```diff
@@ -711,7 +711,7 @@ pub(crate) fn apply_log<
         unsafe { old_forward_queue_length.increment_unchecked(cs) };
     let new_forward_queue_length = UInt32::conditionally_select(
         cs,
-        should_apply,
+        should_apply_io,
```

Comment:
Why does this need to be changed? Btw, maybe I am missing something, but it looks like `should_apply` is equal to true at the beginning if we are processing storage read and then just a few ands performed to `should_apply` and I am not sure that they even can change the value.

Reply:
Decommit opcode doesn't affect IO queue and uses a separate one

Reply:
I see, sorry, misread and thought that `should_apply` equals `boolean_for_variant(STORAGE_READ_OPCODE)`, but now it totally makes sense